httpservice/app/app.py

```diff
@@ -42,20 +44,46 @@ def create_tarfile(directory, tar_filename):
 
 @app.post("/changemaps/", dependencies=[fastapi.Security(check_apikey_header)])
 async def make_changemaps(
-    feacoll: Any = fastapi.Body(), output_tar=fastapi.Depends(sendable_tempfile)
+    data: dict = fastapi.Body(...),
+    output_tar=fastapi.Depends(sendable_tempfile)
 ):
+    input_geojson = data.get("input_geojson")
+    input_t0 = data.get("input_t0")
+    input_t1 = data.get("input_t1")
+
     with tempfile.NamedTemporaryFile(
         "w+", prefix="input-", suffix=".json"
     ) as input_fp, tempfile.TemporaryDirectory() as outdir:
-        json.dump(feacoll, input_fp)
+
+        # Dump feacoll to temp file
+        json.dump(input_geojson, input_fp)
         input_fp.flush()
         input_fp.seek(0)
 
-        gccd.flow(input_fp.name, outdir, "output")
+        # Decode and save input_t0 and input_t1 (if provided)
+        if input_t0:
+            input_t0 = save_base64_to_tempfile(input_t0_base64, suffix=".tiff")
+
+        if input_t1:
```

Comment:
It shouldn't be allowed that t0 is present but not t1 (or vice versa).

I actually might collapse them into a single argument, both at the top-level of the POST body, and also a single argument to gccd.flow.  That one argument can be null/empty/missing, or it can be present.

```
POST /changemaps
{
    "input_geojson": {},
    "images": {
        "t0": "SGVsbG8gd29ybGQh",
        "t1": "Zm9v"
    }
}
```

or when not provided:

```
POST /changemaps
{
    "input_geojson": {}
}
```